src/components/NavBar/store.js

```diff
@@ -1,20 +1,80 @@
-import { observable, action } from 'mobx';
+import { observable, action, computed } from 'mobx';
+import { OracleStatus, Token } from 'constants';
+import _ from 'lodash';
+import Topic from '../../stores/models/Topic';
+import Oracle from '../../stores/models/Oracle';
+
+import { queryAllTopics, queryAllOracles, queryAllVotes } from '../../network/graphQuery';
 
 const INIT_VALUES = {
   myActivitesCount: 0, // INIT_VALUESial loaded state
 };
 
 
-export default class NavBarStore {
-  @observable myActivitesCount = INIT_VALUES.myActivitesCount
+export default class {
+  @observable ResultSettingCount = INIT_VALUES.ResultSettingCount
+  @observable FinalizeCount = INIT_VALUES.FinalizeCount
+  @observable WithdrawCount = INIT_VALUES.WithdrawCount
+  @computed get totalCount() {
+    return this.ResultSettingCount + this.FinalizeCount + this.WithdrawCount;
+  }
 
-  @action.bound
-  async init() {
-    // this.myActivitesCount = await this.getMyActivitiesCount();
+  constructor(app) {
+    this.app = app;
   }
 
-  // @action.bound
-  // async getMyActivitiesCount() {
+  @action
+  init() {
+    this.getCount();
+  }
+
+  @action.bound
+  async getCount() {
+    try {
+      const voteFilters = [];
+      const topicFilters = [];
 
-  // }
+      // Get all votes for all your addresses
+      _.each(this.app.wallet.addresses, (item) => {
+        voteFilters.push({ voterQAddress: item.address });
+        topicFilters.push({ status: OracleStatus.WITHDRAW, creatorAddress: item.address });
+      });
+
+      // Filter votes
+      let votes = await queryAllVotes(voteFilters);
+      votes = votes.reduce((accumulator, vote) => {
+        const { voterQAddress, topicAddress, optionIdx } = vote;
+        if (!_.find(accumulator, { voterQAddress, topicAddress, optionIdx })) accumulator.push(vote);
+        return accumulator;
+      }, []);
+
+      _.each(votes, ({ topicAddress, optionIdx }) => {
+        topicFilters.push({ status: OracleStatus.WITHDRAW, address: topicAddress, resultIdx: optionIdx });
+      });
+      let topics = await queryAllTopics(topicFilters);
+      topics = _.uniqBy(topics, 'txid').map((topic) => new Topic(topic, this.app));
```

Comment:
`uniqBy` should not be necessary here. There should not be any duplicate Topics in any case. If there is, we should actually let the error show up so we can know its happening. But it should never happen anyways.

Also, no need to map it to a `Topic` obj. We are only using the length of the array, not the actual object.

Reply:
hm, I simply follow the logic from the old saga, are you sure we don't need it? I join the team after, sometimes just cannot rethink everything, and if there are no unique thing, it's db problem, should not bother user, we should save it to backlog?

Reply:
@amazingandyyy It would be a DB problem if there are duplicates, which we handle. But, we should assume that by the time it hits the UI that there wont be duplicates and thus no need to filter out with `uniqBy`. We basically assert that there won't be duplicates when we pull the data from the BE.

It's like saying, why don't we do undefined checks for every value coming from the BE even though we know they will be defined.